File: problem-2/main.ex
defmodule Fibonacci do
    def fib(0), do: 1
    def fib(1), do: 1
    def fib(n), do: fib(n-1) + fib(n-2)
end
IO.puts Enum.sum(Enum.filter(
    Enum.take_while(
        Enum.map(1..36, fn x -> Fibonacci.fib(x) end),fn x -> x < 4000000 end), fn x -> rem(x,2) == 0 end
        )
    )

File: problem-2/main.py
def fibonacci(n):
    if n in [0,1]:
        return 1
    else:
        return(fibonacci(n-1)+fibonacci(n-2))
s = 0
i = 1
while True:
    x = fibonacci(i)
    if x > 4_000_000:
        break
    if x % 2 == 0:
        s += x
    i += 1
print(s)
